src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -112,7 +112,7 @@ public override void OnStartActivity(Activity activity, object payload)
 
             if (activity.IsAllDataRequested)
             {
-                activity.SetCustomProperty(RequestCustomPropertyName, request);
+                this.options.Enrich?.Invoke(activity, "OnStartActivity", request);
```

Comment:
Also - can you ensure that unit test validate that Enrich is called only if Activity is sampled-in.

Reply:
Just added a test. Let me know what do you think before I update everything